### PR TITLE
Add SubnetworkInfo, NetworkId, SubnetworkId and test classes

### PR DESCRIPTION
**Pull request by mziccard**

### CODE REVIEW

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/NetworkId.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  /**
+   * Returns the name of the network. The network name must be 1-63 characters long and comply with
+   * RFC1035. Specifically, the name must match the regular expression
+   * {@code [a-z]([-a-z0-9]*[a-z0-9])?} which means the first character must be a lowercase letter,
+   * and all following characters must be a dash, lowercase letter, or digit, except the last
+   * character, which cannot be a dash.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  public final String network() {
```
Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/NetworkId.java]: Codacy Issue found: [Unnecessary final modifier in final class](https://www.codacy.com/app/mziccard/gcloud-java/file/2418453243/issues/source?bid=3140968&fileBranchId=3228978#l57)

File: gcloud-java-compute/src/main/java/com/google/gcloud/compute/SubnetworkInfo.java [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(id, creationTimestamp, subnetworkId, description, gatewayAddress, network,
+        ipRange);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj != null
```
Comment (aozarov) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/SubnetworkInfo.java]: don't you need the `obj == this` to support mocks?

Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/SubnetworkInfo.java]: There's no need for it here, it's needed only when the method is final

Comment (mziccard) [inline gcloud-java-compute/src/main/java/com/google/gcloud/compute/SubnetworkInfo.java]: But I'll add it anyway for consistency.

**Pull request merged**